### DISCUSSION TITLE
Using dynamic import for a component doesn't work as expected

### DISCUSSION THREAD

**Discussion by damiangene** [vercel/next.js#16386] [Help]
I'm running Next.js 9.5.2 with Typescript.
I'm importing and implementing a component via:
```js
const myComponent = dynamic(
  () => import('../components/myComponent').then(mod => mod.myComponent),
  { 
    ssr: false ,
  }
)

export default function Home() {
return (
    <div>
      <Head>
        <title>My site</title>
      </Head>
      <main>
          <myComponent name="Damian" age={27} />
      </main>
    </div>
  )
}
```

However, this won't build and I receive the error:
```Type '{ name: string; age: number; }' is not assignable to type 'IntrinsicAttributes & { children?: ReactNode; }'.```

I've also tried writing the import like this:
```js
const MyComponent = dynamic(async () => {
  const module = await import('components/MyComponent')
  return module.MyComponent
}, { ssr: false })
```
as per another users suggestion with no luck.

So from what I can tell it hasn't imported my component properly as I don't seem to have access to the properties that exist on that component.

This component exists as part of a npm package that I maintain. I've imported it via a dynamic import as it contains references to window and these would throw separate errors if I were to import the component normally.

My page contains other components that I've imported from the npm package via a normal:
```import { ContentHolder, Body } from '../components/containers';```
These work with no issues.

**Comment by lucasromerodb** [comment]
Same problem here! 

```js
// import definition
const DynamicComponent = dynamic(() => import(`@module/components/${fileName}.tsx`));

// use
<DynamicComponent fill={fill} />
```

![image](https://user-images.githubusercontent.com/5304112/102672802-a11b5e00-4170-11eb-9003-4ac9635cd4cf.png)

> Type '{ fill: string; }' is not assignable to type 'IntrinsicAttributes & { children?: ReactNode; }'.
  Property 'fill' does not exist on type 'IntrinsicAttributes & { children?: ReactNode; }'.
  
Have you solve it @damiangene ?

**Reply by tinavanschelt** [reply]
@lucasromerodb, I managed to fix this with:

```typescript
// define type
type Foo = {
  fill: string;
};

// import definition
const DynamicComponent = dynamic<Foo>(() => import(`@module/components/${fileName}.tsx`));

// use
<DynamicComponent fill={fill} />
```

Hope it helps!
